scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -856,14 +874,24 @@ object Execution {
    * these executions are executed in parallel: run is called on all at the
    * same time, not one after the other.
    */
+  private case class SequencingFn[T]() extends Function1[(T, List[T]), List[T]] {
```

Comment:
I see, yours actually does the same as mine without casting due to the way equals works on case classes.

That's better (casts are ugly).